hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -370,33 +335,16 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
             totalCorruptBlocks.incrementAndGet();
             // If there is a corrupt block - we will assume that this was the next data block
             currentInstantLogBlocks.push(logBlock);
-            validLogBlockInstants.add(logBlock);
-            // we don't need to update the block sequence tracker here, since the block sequence tracker is meant to remove additional/spurious valid logblocks.
-            // anyway, contents of corrupt blocks are not read.
             break;
           default:
             throw new UnsupportedOperationException("Block type not supported yet");
         }
       }
       // merge the last read block when all the blocks are done reading
       if (!currentInstantLogBlocks.isEmpty()) {
-        boolean duplicateBlocksDetected = false;
-        if (blockIdentifiersPresent.get()) {
-          Pair<Boolean, List<HoodieLogBlock>> dedupedLogBlocksInfo = reconcileSpuriousBlocksAndGetValidOnes(validLogBlockInstants, blockSequenceMapPerCommit);
-          duplicateBlocksDetected = dedupedLogBlocksInfo.getKey();
-          if (duplicateBlocksDetected) {
-            // if there are duplicate log blocks that needs to be removed, we re-create the queue for valid log blocks from dedupedLogBlocks
-            currentInstantLogBlocks = new ArrayDeque<>();
-            dedupedLogBlocksInfo.getValue().forEach(block -> currentInstantLogBlocks.push(block));
-            LOG.info("Merging the final data blocks");
-            processQueuedBlocksForInstant(currentInstantLogBlocks, scannedLogFiles.size(), keySpecOpt);
-          }
-        }
-        if (!duplicateBlocksDetected) {
-          // if there are no dups, we can take currentInstantLogBlocks as is.
-          LOG.info("Merging the final data blocks");
-          processQueuedBlocksForInstant(currentInstantLogBlocks, scannedLogFiles.size(), keySpecOpt);
-        }
+        // if there are no dups, we can take currentInstantLogBlocks as is.
+        LOG.info("Merging the final data blocks");
+        processQueuedBlocksForInstant(currentInstantLogBlocks, scannedLogFiles.size(), keySpecOpt);
```

Comment:
Changes around `HoodieAppendHandle#addBlockIdentifier`, `TaskContextSupplier#getAttemptNumberSupplier` introduced by #9611 should also be reverted.

Reply:
we added additional apis to taskContextSupplier. I might be using them in a follow up patch. So, left it as is. but there is no caller to those apis right since we reverted them?

Reply:
OK.  My intention is to make sure no one else accidentally uses the APIs to cause more issues.